test/test.sh

```
@@ -73,5 +123,49 @@ assert "echo \"'hello	world'\" \"42Tokyo\""
+export	EOF='eof'
+assert	'cat <<$EOF			\neof\nEOF\nEOF'
+assert	'cat <<"$EOF		\neof\nEOF\nEOF'
+
+# Pipe
+assert 'cat Makefile | grep minishell'
+assert 'cat | cat | ls\n\n'
+
+# Expand Variable
+assert 'echo $USER'
+assert 'echo $USER$PATH&TERM'
+assert'echo "$USER $PATH $TERM"'
```

Shellcheck (0.11.0)

[warning] This is interpreted as a command name ending with doublequote. Double check syntax.

(SC2288)